src/tools/miri/tests/pass/weak_memory/weak.rs

```diff
@@ -33,69 +33,67 @@ fn reads_value(loc: &AtomicUsize, val: usize) -> usize {
 
 fn relaxed(initial_read: bool) -> bool {
     let x = static_atomic(0);
-    let j1 = spawn(move || {
-        x.store(1, Relaxed);
-        // Preemption is disabled, so the store above will never be the
-        // latest store visible to another thread.
-        x.store(2, Relaxed);
-    });
+    let j = spawn(move || x.load(Relaxed));
 
-    let j2 = spawn(move || x.load(Relaxed));
+    x.store(1, Relaxed);
+    // Preemption is disabled, so the store above will never be the
+    // latest store visible to the other thread.
+    x.store(2, Relaxed);
 
-    j1.join().unwrap();
-    let r2 = j2.join().unwrap();
+    let r = j.join().unwrap();
 
     // There are three possible values here: 0 (from the initial read), 1 (from the first relaxed
     // read), and 2 (the last read). The last case is boring and we cover the other two.
-    r2 == if initial_read { 0 } else { 1 }
+    r == if initial_read { 0 } else { 1 }
 }
 
 // https://www.doc.ic.ac.uk/~afd/homepages/papers/pdfs/2017/POPL.pdf Figure 8
-fn seq_cst() -> bool {
+fn seq_cst(relaxed_write: bool) -> bool {
```

Comment:
This argument is only used to determine the expected result, it doesn't actually change what the test does...?